.github/workflows/build_and_run_apps.yml

```diff
@@ -0,0 +1,147 @@
+name: Build and Run Apps
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Once per day at midnight
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    name: Build Apps
+    strategy:
+      fail-fast: false
+      matrix:
+        idf_ver: 
+          - "release-v5.0"
+          - "release-v5.1"
+          - "release-v5.2"
+          - "release-v5.3"
+          - "latest"
+        parallel_index: [1,2,3,4,5] # Update --parallel-count below when changing this
+    runs-on: ubuntu-22.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - name: Install dependencies
+        shell: bash
+        run: |
+          . ${IDF_PATH}/export.sh
+          pip install --upgrade idf-component-manager 'idf-build-apps>=2.4,<2.5'
```

Comment:
now 2.5.2 released. Please have a try.

Reply:
Thanks! I'll do this in the next MR, in this one the pipeline already passed, I'll take the opportunity to finally merge it.